### ISSUE TITLE
`HLint` warns when it shouldn't  

### ISSUE THREAD

**Issue by vrom911** [ndmitchell/hlint#448]
I have the following code:
```haskell
import Universum

import Control.Monad.Except (MonadError (..))

foo = runExceptT
```
And I have this `.hlint.yaml`
```yaml
- warn: { name: "Use 'runExceptT' from Universum"
        , lhs: Control.Monad.Except.runExceptT, rhs: Universum.runExceptT }
```
When I run `hlint` I get:
```bash
$ hlint example.hs 
example.hs:5:7: Warning: Use 'runExceptT' from Universum
Found:
  runExceptT
Why not:
  runExceptT

1 hint
```
But the thing is that `runExceptT` is exported from `Universum` here, because `MonadError` class doesn't even have `runExceptT` method, but `hlint` (probably because of wildcards) assumes that `runExceptT` is used from `Control.Monad.Except` module.
If I remove these `(..)` from import then `hlint doesn't throws any warnings.

I'm on `2.0.11` version of `hlint`.

**Comment by ChShersh** [comment]
As a custom prelude maintainer I find this issue rather annoying... But I would love to help with it if somebody can point me to relevant pieces of code! Maybe @vrom911 also willing to help.

**Comment by vrom911** [comment]
Yes, I definitely would like to help with this if @ndmitchell doesn't mind having these changes.

**Comment by ndmitchell** [comment]
I'm not sure it's easy to solve this, or even possible. HLint doesn't know where runExceptT comes from - it might be Universum or might be Control.Monad.Except, so it says it matches.

If you remove the `(..)` then HLint knows for sure, and then doesn't warn.

**Comment by ChShersh** [comment]
@ndmitchell One solution is to use `haskell-names` library to determine what entities came from each import:

* https://github.com/haskell-suite/haskell-names

I've used this library during my work on `importify`:

* https://github.com/serokell/importify

You're right, this is not an easy solution at all. But it should be possible.

**Comment by ndmitchell** [comment]
haskell-names requires tracing dependencies, and in many cases hlint doesn't have the dependencies available. Wiring them in would be a lot of work, but also break cases where HLint currently works on a standalone file with no dependencies installed.
